src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,14 +1,45 @@
-import React from 'react';
+import React, { ChangeEvent, useCallback } from 'react';
+import { useDispatch } from 'react-redux';
+import { useAppSelector } from '../../app/hooks';
+import { actions as filterActions } from '../../features/filter';
+import { Status } from '../../types/Status';
 
 export const TodoFilter: React.FC = () => {
+  const { query, status } = useAppSelector(state => state.filter);
+  const dispatch = useDispatch();
+
+  const handleSelect = useCallback(
+    (event: ChangeEvent<HTMLSelectElement>) => {
+      const value = event.target.value as Status;
+
+      dispatch(filterActions.setStatus(value));
+    },
+    [],
+  );
+
+  const handleInput = useCallback(
+    (event: ChangeEvent<HTMLInputElement>) => {
+      dispatch(filterActions.setQuery(event.target.value));
+    },
+    [],
+  );
+
+  const handleClearQuery = useCallback(() => {
+    dispatch(filterActions.clearQuery());
+  }, []);
```

Comment:
Don't use `useCallback` without a reason. It's even a little bit worse for the performance this way.

Here you can read more about it. If you have more questions about it - feel free to ask in the chat.
https://react.dev/reference/react/useCallback